sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -398,6 +398,51 @@ def test_one_hot_encoder_feature_names_drop(drop, expected_names):
     assert_array_equal(expected_names, feature_names)
 
 
+@pytest.mark.parametrize(
+    "X, expected, expected_drop_idx",
+    [
+        (
+            [['Male', 1],
+             ['Female', 3],
+             ['Female', 2]],
+
+            np.array([[1., 1., 0., 0.],
+                      [0., 0., 0., 1.],
+                      [0., 0., 1., 0.]]),
+
+            np.array([0, -1]),
+        ),
+        (
+            [[10, 'yes'],
+             [20, 'no'],
+             [30, 'yes']],
+
+            np.array([[1., 0., 0., 1.],
+                      [0., 1., 0., 0.],
+                      [0., 0., 1., 1.]]),
+
+            np.array([-1, 0]),
+        ),
+        (
+            [['true', 'a'],
+             ['false', 'a'],
+             ['false', 'a']],
+
+            np.array([[1., 1.],
+                      [0., 1.],
+                      [0., 1.]]),
+
+            np.array([0, -1]),
```

Comment:
We don't need to parametrize, as it's not very readable. We can just write these 3 cases one after the other in the test.

Reply:
done and all greens